### PR TITLE
Automate things

### PR DESCRIPTION
**Pull request by UziTech**
This is a rebase of #8

### CODE REVIEW

**Comment by the-j0k3r** [comment]
Is this ready to go in? Do I need to install any bots?

File: package.json [hunk 1]
```diff
@@ -56,15 +56,30 @@
     "console",
     "powershell"
   ],
+  "scripts": {
+    "test": "atom --test spec",
+    "lint": "eslint .",
```
Comment (the-j0k3r) [inline package.json]: it would probably be good to add a script with --fix later on.

**Comment by UziTech** [comment]
> Do I need to install any bots?

Yes, we will need to setup the following:

- [appveyor](https://ci.appveyor.com/)
- [renovate](https://github.com/marketplace/renovate)
- [travis ci](https://travis-ci.com)
	- Copy API token from https://atom.io/account
	- Set `ATOM_ACCESS_TOKEN` in travis [settings](https://travis-ci.org/bus-stop/terminus/settings)
		image
	- Create a GitHub personal access token https://github.com/settings/tokens
	- Just check public-repo
		image
	- Set `GH_TOKEN` in travis settings
		image

**Comment by UziTech** [comment]
I think this is ready to go we have:

- automated testing in linux, macos, and windows with stable and beta builds of Atom
- automated dependency management with renovate. devdependencies will auto merge after testing but regular dependencies will just create a PR and let us merge if we know it won't break anything
- automated publishing with semantic-release
- linting with eslint will be tested on every PR

**Comment by the-j0k3r** [comment]
Will appveyor need any tokens? Travis is done, renovate is done.

**Comment by UziTech** [comment]
no appveyor doesn't need anything

**Comment by the-j0k3r** [comment]
Thanks,

All is building ATM, AV done Travis OTW

**Comment by UziTech** [comment]
Now the hard part will be actually writing tests

**Comment by the-j0k3r** [comment]
Sent have an invite for the org, use as you will.

> Now the hard part will be actually writing tests

For me almost all aspects of developing are hard, but learning as I go.
Aside from writing tests. theres the actual valid legacy bugs, and using deps that arent abandoned.

You mentioned `Space pen` and we also have the infamous `term.js` also which the later would probably get rid of most legacy bugs.

ATM looks good except shortcuts dont work so well in Linux for e.g `ctrl`+`'` has no effect. thats a legacy bug right there too.

Ah well, Rome wasnt built in a day. =)

**Pull request merged**